### PR TITLE
Add changelog and update readme for external-resizer

### PR DESCRIPTION
**Pull request by gnufied**
Add changelog for 1.4 release and update readme.

### CODE REVIEW

**Review by bswartz — APPROVED** [review]
/lgtm

File: README.md [hunk 1]
```diff
@@ -77,6 +77,14 @@ Note that the external-resizer does not scale with more replicas. Only one exter
   * `AnnotateFsResize=true|false` (ALPHA - default=false): Store current size of pvc in pv's annotation, so as if pvc is deleted while expansion was pending on the node, the size of pvc can be restored to old value. This permits
     expansion on the node in case pvc was deleted while expansion was pending on the node (but completed in the controller). Use of this feature depends on Kubernetes version 1.21.
 
+  * `RecoverVolumeExpansionFailure=true|false` (ALPHA - default=false): Allow users to reduce size of PVC if expansion to current size is failing. If the feature gate `RecoverVolumeExpansionFailure` is enabled
+    and expansion has failed for a PVC, you can retry expansion with a smaller size than the previously requested value. To request a new expansion attempt with a
```
Comment (jingxu97) [inline README.md:81]: but the new requested value must be larger than the original one, right?

Comment (gnufied) [reply README.md:81]: Sorry it got merged but yes. This is documented here - https://kubernetes.io/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims

**Pull request merged**